src/genesis/observers/balances.py

```diff
@@ -64,13 +68,60 @@ def __init__(self, db_conn: Connection, on_completed=None, on_error=None) -> Non
                                                 on_completed=on_completed,
                                                 on_error=on_error)
 
+    def _get_name_and_index(self, e: UniqueViolation, balances: List[Balance]) -> Tuple[str, Tuple[int, int]]:
+        # Extract account name and coin from error string
+        duplicate_balance = str(e).split("(")[2].split(")")[0]
```

Comment:
I'm confused about the data structure here. Can we do something to clarify, is this part of a JSON string or something else?

Reply:
If this is just the ID then I would propose renaming it to `duplicate_balance_id` to be consistent with what you had done with `account_id`.